domain/user/service/service.go

```diff
@@ -233,26 +301,31 @@ func (s *Service) RemoveUser(ctx context.Context, name string) error {
 // - internal/auth.ErrPasswordNotValid: If the password supplied is not valid.
 func (s *Service) SetPassword(
 	ctx context.Context,
-	name string,
+	uuid user.UUID,
 	password auth.Password,
 ) error {
 	defer password.Destroy()
-	if err := ValidateUsername(name); err != nil {
-		return fmt.Errorf("username %q: %w", name, err)
+	if err := uuid.Validate(); err != nil {
+		return fmt.Errorf("setting password for user with id %q: %w", uuid, err)
+	}
+
+	usr, err := s.GetUser(ctx, uuid)
+	if err != nil {
+		return fmt.Errorf("setting password for user %q: %w", usr.Name, err)
 	}
 
 	salt, err := auth.NewSalt()
 	if err != nil {
-		return fmt.Errorf("setting password for user %q, generating password salt: %w", name, err)
+		return fmt.Errorf("setting password for user %q, generating password salt: %w", usr.Name, err)
```

Comment:
I think it's great that we are include the user name in the error message but I am wondering if we should also be including the UUID?

My thinking is our job with error message to is to annotate them with as much context as we can. Considering all of these functions are getting called with uuid as the context we should be providing that back up.

Just a random thought. Applies to all the errors in this file where UUID is around.

Reply:
good idea,

may be we can do something like with `<user_name>(<user_uuid>)`